### ISSUE TITLE
Error provider injection when launching an agent using Janus kernel from code in Eclipse

### ISSUE THREAD

**Issue by ngaud** [sarl/sarl#400]
You obtained the following Error when you try to do this 
```
	    			Boot.setOffline(true);
	    			Boot.setVerboseLevel(LoggerCreator.toInt(Level.INFO));
	    			Boot.showJanusLogo();
	    			try {
						Boot.startJanus(
								(Class) null,
								Environment.class,
								AntGUI.this.getG(),
								AntGUI.this);
					} catch (Exception e1) {
						// TODO Auto-generated catch block
						e1.printStackTrace();
					}
	    	    	
```

```
janv. 13, 2016 4:33:51 PM null
INFOS: Contexte d'agent créé : 2c38fb7f-f363-4f6e-877b-110b1f07cc77
janv. 13, 2016 4:33:51 PM null
INFOS: Espace créé : SpaceID [id=7ba8885d-545b-445a-a0e9-b655bc15ebe0, contextID=2c38fb7f-f363-4f6e-877b-110b1f07cc77, spaceSpec=interface io.sarl.util.OpenEventSpaceSpecification]
janv. 13, 2016 4:33:51 PM io.janusproject.kernel.services.arakhne.ArakhneLocaleLogService
INFOS: Lancement de l'agent : ACO_agent.Environment
io.janusproject.kernel.services.jdk.spawn.StandardSpawnService$CannotSpawnException: Impossible d'instancier un agent de type class ACO_agent.Environment à cause de: Unable to provision, see the following errors:

1) Error in custom provider, io.janusproject.kernel.services.jdk.spawn.StandardSpawnService$CannotSpawnException: Impossible d'instancier un agent de type class ACO_agent.Environment à cause de: null
  at io.janusproject.kernel.services.jdk.spawn.StandardSpawnService$JustInTimeAgentInjectionModule.configure(StandardSpawnService.java:468)
  while locating io.sarl.lang.core.Agent

1 error
	at io.janusproject.kernel.services.jdk.spawn.StandardSpawnService.spawn(StandardSpawnService.java:131)
	at io.janusproject.kernel.Kernel.spawn(Kernel.java:142)
	at io.janusproject.Boot.startJanus(Boot.java:681)
	at io.janusproject.Boot.startJanus(Boot.java:640)
	at UI.AntGUI$2.handle(AntGUI.java:163)
	at UI.AntGUI$2.handle(AntGUI.java:1)
	at com.sun.javafx.event.CompositeEventHandler.dispatchBubblingEvent(CompositeEventHandler.java:86)
	at com.sun.javafx.event.EventHandlerManager.dispatchBubblingEvent(EventHandlerManager.java:238)
	at com.sun.javafx.event.EventHandlerManager.dispatchBubblingEvent(EventHandlerManager.java:191)
	at com.sun.javafx.event.CompositeEventDispatcher.dispatchBubblingEvent(CompositeEventDispatcher.java:59)
	at com.sun.javafx.event.BasicEventDispatcher.dispatchEvent(BasicEventDispatcher.java:58)
	at com.sun.javafx.event.EventDispatchChainImpl.dispatchEvent(EventDispatchChainImpl.java:114)
	at com.sun.javafx.event.BasicEventDispatcher.dispatchEvent(BasicEventDispatcher.java:56)
	at com.sun.javafx.event.EventDispatchChainImpl.dispatchEvent(EventDispatchChainImpl.java:114)
	at com.sun.javafx.event.BasicEventDispatcher.dispatchEvent(BasicEventDispatcher.java:56)
	at com.sun.javafx.event.EventDispatchChainImpl.dispatchEvent(EventDispatchChainImpl.java:114)
	at com.sun.javafx.event.EventUtil.fireEventImpl(EventUtil.java:74)
	at com.sun.javafx.event.EventUtil.fireEvent(EventUtil.java:49)
	at javafx.event.Event.fireEvent(Event.java:198)
	at javafx.scene.Node.fireEvent(Node.java:8411)
	at javafx.scene.control.Button.fire(Button.java:185)
	at com.sun.javafx.scene.control.behavior.ButtonBehavior.mouseReleased(ButtonBehavior.java:182)
	at com.sun.javafx.scene.control.skin.BehaviorSkinBase$1.handle(BehaviorSkinBase.java:96)
	at com.sun.javafx.scene.control.skin.BehaviorSkinBase$1.handle(BehaviorSkinBase.java:89)
	at com.sun.javafx.event.CompositeEventHandler$NormalEventHandlerRecord.handleBubblingEvent(CompositeEventHandler.java:218)
	at com.sun.javafx.event.CompositeEventHandler.dispatchBubblingEvent(CompositeEventHandler.java:80)
	at com.sun.javafx.event.EventHandlerManager.dispatchBubblingEvent(EventHandlerManager.java:238)
	at com.sun.javafx.event.EventHandlerManager.dispatchBubblingEvent(EventHandlerManager.java:191)
	at com.sun.javafx.event.CompositeEventDispatcher.dispatchBubblingEvent(CompositeEventDispatcher.java:59)
	at com.sun.javafx.event.BasicEventDispatcher.dispatchEvent(BasicEventDispatcher.java:58)
	at com.sun.javafx.event.EventDispatchChainImpl.dispatchEvent(EventDispatchChainImpl.java:114)
	at com.sun.javafx.event.BasicEventDispatcher.dispatchEvent(BasicEventDispatcher.java:56)
	at com.sun.javafx.event.EventDispatchChainImpl.dispatchEvent(EventDispatchChainImpl.java:114)
	at com.sun.javafx.event.BasicEventDispatcher.dispatchEvent(BasicEventDispatcher.java:56)
	at com.sun.javafx.event.EventDispatchChainImpl.dispatchEvent(EventDispatchChainImpl.java:114)
	at com.sun.javafx.event.EventUtil.fireEventImpl(EventUtil.java:74)
	at com.sun.javafx.event.EventUtil.fireEvent(EventUtil.java:54)
	at javafx.event.Event.fireEvent(Event.java:198)
	at javafx.scene.Scene$MouseHandler.process(Scene.java:3757)
	at javafx.scene.Scene$MouseHandler.access$1500(Scene.java:3485)
	at javafx.scene.Scene.impl_processMouseEvent(Scene.java:1762)
	at javafx.scene.Scene$ScenePeerListener.mouseEvent(Scene.java:2494)
	at com.sun.javafx.tk.quantum.GlassViewEventHandler$MouseEventNotification.run(GlassViewEventHandler.java:352)
	at com.sun.javafx.tk.quantum.GlassViewEventHandler$MouseEventNotification.run(GlassViewEventHandler.java:275)
	at java.security.AccessController.doPrivileged(Native Method)
	at com.sun.javafx.tk.quantum.GlassViewEventHandler.lambda$handleMouseEvent$354(GlassViewEventHandler.java:388)
	at com.sun.javafx.tk.quantum.QuantumToolkit.runWithoutRenderLock(QuantumToolkit.java:389)
	at com.sun.javafx.tk.quantum.GlassViewEventHandler.handleMouseEvent(GlassViewEventHandler.java:387)
	at com.sun.glass.ui.View.handleMouseEvent(View.java:555)
	at com.sun.glass.ui.View.notifyMouse(View.java:937)
	at com.sun.glass.ui.gtk.GtkApplication._runLoop(Native Method)
	at com.sun.glass.ui.gtk.GtkApplication.lambda$null$49(GtkApplication.java:139)
	at java.lang.Thread.run(Thread.java:745)
Caused by: com.google.inject.ProvisionException: Unable to provision, see the following errors:

1) Error in custom provider, io.janusproject.kernel.services.jdk.spawn.StandardSpawnService$CannotSpawnException: Impossible d'instancier un agent de type class ACO_agent.Environment à cause de: null
  at io.janusproject.kernel.services.jdk.spawn.StandardSpawnService$JustInTimeAgentInjectionModule.configure(StandardSpawnService.java:468)
  while locating io.sarl.lang.core.Agent

1 error
	at com.google.inject.internal.InjectorImpl$2.get(InjectorImpl.java:1025)
	at com.google.inject.internal.InjectorImpl.getInstance(InjectorImpl.java:1051)
	at io.janusproject.kernel.services.jdk.spawn.StandardSpawnService.spawn(StandardSpawnService.java:125)
	... 52 more
Caused by: io.janusproject.kernel.services.jdk.spawn.StandardSpawnService$CannotSpawnException: Impossible d'instancier un agent de type class ACO_agent.Environment à cause de: null
	at io.janusproject.kernel.services.jdk.spawn.StandardSpawnService$JustInTimeAgentInjectionModule.get(StandardSpawnService.java:480)
	at io.janusproject.kernel.services.jdk.spawn.StandardSpawnService$JustInTimeAgentInjectionModule.get(StandardSpawnService.java:446)
	at com.google.inject.internal.ProviderInternalFactory.provision(ProviderInternalFactory.java:81)
	at com.google.inject.internal.InternalFactoryToInitializableAdapter.provision(InternalFactoryToInitializableAdapter.java:53)
	at com.google.inject.internal.ProviderInternalFactory.circularGet(ProviderInternalFactory.java:61)
	at com.google.inject.internal.InternalFactoryToInitializableAdapter.get(InternalFactoryToInitializableAdapter.java:45)
	at com.google.inject.internal.InjectorImpl$2$1.call(InjectorImpl.java:1016)
	at com.google.inject.internal.InjectorImpl.callInContext(InjectorImpl.java:1092)
	at com.google.inject.internal.InjectorImpl$2.get(InjectorImpl.java:1012)
	... 54 more
Caused by: java.lang.reflect.InvocationTargetException
	at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
	at sun.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
	at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
	at java.lang.reflect.Constructor.newInstance(Constructor.java:422)
	at io.janusproject.kernel.services.jdk.spawn.StandardSpawnService$JustInTimeAgentInjectionModule.get(StandardSpawnService.java:477)
	... 62 more
Caused by: java.lang.NoSuchMethodError: com.google.common.eventbus.EventBus.<init>(Ljava/lang/String;Ljava/util/concurrent/Executor;Lcom/google/common/eventbus/Dispatcher;Lcom/google/common/eventbus/SubscriberExceptionHandler;Ljava/lang/Class;)V
	at com.google.common.eventbus.AsyncSyncEventBus.<init>(AsyncSyncEventBus.java:31)
	at io.janusproject.modules.kernel.MandatoryKernelModule.createAgentInternalBus(MandatoryKernelModule.java:96)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:497)
	at com.google.inject.internal.ProviderMethod$ReflectionProviderMethod.doProvision(ProviderMethod.java:299)
	at com.google.inject.internal.ProviderMethod.get(ProviderMethod.java:172)
	at com.google.inject.internal.ProviderInternalFactory.provision(ProviderInternalFactory.java:81)
	at com.google.inject.internal.InternalFactoryToInitializableAdapter.provision(InternalFactoryToInitializableAdapter.java:53)
	at com.google.inject.internal.ProviderInternalFactory.circularGet(ProviderInternalFactory.java:61)
	at com.google.inject.internal.InternalFactoryToInitializableAdapter.get(InternalFactoryToInitializableAdapter.java:45)
	at com.google.inject.internal.SingleFieldInjector.inject(SingleFieldInjector.java:54)
	at com.google.inject.internal.MembersInjectorImpl.injectMembers(MembersInjectorImpl.java:132)
	at com.google.inject.internal.MembersInjectorImpl$1.call(MembersInjectorImpl.java:93)
	at com.google.inject.internal.MembersInjectorImpl$1.call(MembersInjectorImpl.java:80)
	at com.google.inject.internal.InjectorImpl.callInContext(InjectorImpl.java:1103)
	at com.google.inject.internal.MembersInjectorImpl.injectAndNotify(MembersInjectorImpl.java:80)
	at com.google.inject.internal.MembersInjectorImpl.injectMembers(MembersInjectorImpl.java:62)
	at com.google.inject.internal.InjectorImpl.injectMembers(InjectorImpl.java:984)
	at io.janusproject.kernel.bic.StandardBuiltinCapacitiesProvider.getBuiltinCapacities(StandardBuiltinCapacitiesProvider.java:94)
	at io.sarl.lang.core.Agent.<init>(Agent.java:69)
	at ACO_agent.Environment.<init>(Environment.java:530)
	... 67 more
```

**Comment by ngaud** [comment]
The problem is related to a conflict in Guava version, when you have the SARL Libaries in the classpath of your project, you directly references Google Guava 15.0 insted of using the fork version of guava included within Janus jar.

The solutions consists in removing the SARL librairies from the classpath of your project when you include a direct reference to the Janus Jar in the project classpath.

Basically you should a single refernece to the fork of google guava used in Janus.

**Comment by ngaud** [comment]
An option would be to check in our Eclipse plug-in if we have multiple reference to guava in the classpath and automatically choose the best.

**Comment by gallandarakhneorg** [comment]
I think I better way is to add or remove the SARL libraries when a standalone SRE is removed or added on the classpath. We should not care about the imported libraries by themselves but the fact the standalone SRE is already embedding a specific SARL library.

**Comment by gallandarakhneorg** [comment]
This issue should be solved by issue #314.

**Comment by gallandarakhneorg** [comment]
Since Issue #303 is closed, this issue becomes void.
